desc/transform.py

```diff
@@ -71,29 +71,19 @@ def __init__(
                 )
             )
 
-        self._derivatives = self._get_derivatives(derivs)
-        self._sort_derivatives()
-        self._method = method
-
         self._built = False
         self._built_pinv = False
-        self._set_up()
+        self._derivatives = self._get_derivatives(derivs)
+        self._sort_derivatives()
+        # assign according to logic in setter function
+        self.method = method
+        # assign according to logic in property function
+        self._matrices = self.matrices
```

Comment:
Mixing the design patterns of using setters and property functions to assign attributes is a little confusing. If this is done elsewhere would prefer if comments like this are added to clarify.

Reply:
Not sure what the problem is. This is the error.
> tests/test_perturbations.py::test_perturbation_orders - AttributeError: 'Transform' object has no attribute '_matrices'. Did you mean: 'matrices'?
I've added the following guard to the end of `__init__`. It does not raise an error, so this attribute must be getting deleted somewhere?
```python
        self._matrices = self.matrices
        if not hasattr(self, "_matrices"):
            raise ValueError("Kaya does not know how to python. Uno")
        if build:
            self.build()
        if build_pinv:
            self.build_pinv()

        if not hasattr(self, "_matrices"):
            raise ValueError("Kaya does not know how to python. Dos")
```

Reply:
not totally sure what the issue is, but if you remove the line 
```
        self._matrices = self.matrices
``` 
from init and also replace all the references to ``self._matrices`` elsewhere in the class with ``self.matrices`` it seems to work for me

Reply:
Is the def method setter supposed to be recursive? If you notice it calls itself because it assigns with self.method = rather than self._method. "Fixing" this to make it not recursive generates similar missing attribute errors. It's not clear to me why either of these issues occur from making these changes, but maybe they are related.

I'm hesitant to make your change. Currently it seems the matrices attribute is getting deleted after __init__, and I think your suggestion will avoid that by constantly reconstructing it whenever its called.

Reply:
It seems like it's because I removed the `def _set_up` method, which apparently is used in `desc/io/equilibrium_io.py` for something. If that does end up being the issue, we should document the purpose more clearly. I think it is unusual to rely on a private method like `_set_up`.